src/code_review_agent/webhooks/github_webhook.py

```
@@ -0,0 +1,344 @@
+            return False
+    
+    async def handle_webhook(
+        self,
+        payload: GitHubWebhookPayload,
+        signature: Optional[str] = None,
+        payload_body: Optional[bytes] = None
+    ) -> Dict[str, str]:
+        """Handle incoming GitHub webhook."""
+        # Verify signature if provided
+        if signature and payload_body:
+            if not self.verify_signature(payload_body, signature):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)